src/sbvr-api/sbvr-utils.ts

```diff
@@ -1049,6 +1049,44 @@ export class PinejsClient<
 	public post(params: Params): Promise<AnyObject> {
 		return super.post(params as Parameters<PinejsClient<M>['post']>[0]);
 	}
+	public compileAuth<TResource extends keyof M & string>(params: {
+		modelName: string;
+		access: string;
+		resource: TResource;
+		options?: Pick<ODataOptions<M[TResource]['Read']>, '$filter'>;
+	}): string {
+		const { modelName, access, resource, options } = params;
+		if (typeof modelName !== 'string') {
+			throw new Error(`The modelName property must be specified.`);
+		}
+		if (typeof access !== 'string') {
+			throw new Error(`The access property must be specified.`);
+		}
+		if (typeof resource !== 'string') {
+			throw new Error('The resource must be specified.');
+		}
+
+		const authBase = `${modelName}.${resource}.${access}`;
+		const url =
+			options?.$filter != null ? this.compile({ resource, options }) : null;
+		if (url == null) {
+			return authBase;
+		}
```

Comment:
This might be slightly nicer as
```suggestion
		if (options?.$filter == null) {
			return authBase;
		}
		const url = this.compile({ resource, options });
```
assuming of course it can play nice with typings